lib/Dialect/HW/HWOps.cpp

```diff
@@ -2340,6 +2452,19 @@ void ArrayGetOp::build(OpBuilder &builder, OperationState &result, Value input,
 // the index. If the array is constructed from a constant by a bitcast
 // operation, we can fold into a constant.
 OpFoldResult ArrayGetOp::fold(ArrayRef<Attribute> operands) {
+  if (operands[0]) {
+    auto arrayAttr = operands[0].cast<ArrayAttr>();
+    // Constant array index.
+    if (operands[1]) {
+      auto index = operands[1].cast<IntegerAttr>().getValue();
+      return arrayAttr[index.getZExtValue()];
```

Comment:
I guess `arrayAttr[arrayAttr.size() - 1 - index.getZExtValue()];` should be correct here.

Reply:
Thanks!